src/redprl/refiner_misc.fun

```diff
@@ -192,13 +193,13 @@ struct
             Metavar.Ctx.empty
             (metas, args)
 
-
         val ty' = substMetaenv mrho ty
-        val _ = if Hyps.isEmpty H' then () else raise Fail "Synth.Custom only works with empty sequent"
+        val _ = if Hyps.isEmpty H' then () else
+          E.raiseError @@ E.IMPOSSIBLE (Fpp.text "Open judgments attached to custom operator.")
```

Comment:
@cangiuli @jonsterling Any suggestion on the error message about somehow a sequent with non-empty `H` being associated with a custom operator? (It should be impossible, because the parser should prevent the user from doing so.)

@jonsterling Actually, is there a reason not to forbid `H` directly in typing?

Reply:
@favonia I need to think about this a little bit. Don't worry about it for now, but I can hopefully make this a lot cleaner.